pulp_rpm/app/models/repository.py

```diff
@@ -447,9 +447,9 @@ def _apply_retention_policy(self, new_version):
                 Package.objects.with_age()
                 .filter(
                     pk__in=new_version.content.filter(pulp_type=Package.get_pulp_type()),
-                    is_modular=False,  # don't want to filter out modular RPMs
-                )
-                .only("pk")
+                ).exclude(
+                    pk__in=pulp_rpm.app.models.modulemd.Modulemd_packages.values_list("package", flat=True),
+                ).only("pk")
```

Comment:
This on the other hand might be tricky to do in a performant way.  The through table is implicit so this doesn't actually work.

Reply:
is it worth filtering modulemd within the current repo version?